File: engine/transpile/src/test/resources/golden/SYK001/VALID_REC.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK001 のデータ部レコード VALID-REC を逐語対訳した自動生成コード。 */
public final class VALID_REC {

    /** 01 VALID-REC 総バイト長 253。 */
    public static final int LENGTH = 253;

    private final byte[] data;

    public VALID_REC() {
        this.data = new byte[253];
    }

    public VALID_REC(byte[] source) {
        this.data = Arrays.copyOf(source, 253);
    }

    public byte[] data() {
        return data;
    }

    // VALID-REC
    public String get_VALID_REC() {
        return CobolRuntime.decodeAlphanumeric(data, 0, 253);
    }
    public void set_VALID_REC(String value) {
        CobolRuntime.encodeAlphanumeric(data, 0, 253, value);
    }
}

File: src/engine/transpile/src/test/resources/golden/SYK001/VALID_REC.py
"""SYK001 のデータ部レコード VALID-REC を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class VALID_REC:
    """01 VALID-REC 総バイト長 253。"""

    LENGTH = 253

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(253)
        else:
            self._data = bytearray(data)
            if len(self._data) < 253:
                self._data.extend(b"\x00" * (253 - len(self._data)))

    @property
    def data(self):
        return self._data

    # VALID-REC
    def get_VALID_REC(self):
        return decode_alphanumeric(self._data, 0, 253)
    def set_VALID_REC(self, value):
        encode_alphanumeric(self._data, 0, 253, value)
